compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -1833,15 +1834,15 @@ public void setInputClauseTypedBindingPatternType(BLangInputClause bLangInputCla
         }
 
         BType collectionType = bLangInputClause.collection.getBType();
-        BType varType = visitCollectionType(bLangInputClause, collectionType);
-        if (varType.tag == TypeTags.SEMANTIC_ERROR) {
+        bLangInputClause.varType = visitCollectionType(bLangInputClause, collectionType);
+        if (bLangInputClause.varType.tag == TypeTags.SEMANTIC_ERROR || collectionType.tag == OBJECT) {
```

Comment:
What was the reason for filtering objects here?

Reply:
Because for objects if it is valid we set `bLangInputClause` properties inside `visitCollectionType` method itself.